### PR TITLE
chore: don't allow editing of a task if it is backed by a script

### LINKED ISSUE #4978: Fix tasks edit experience for tasks that invoke scripts
The task edit experience in the UI requires some changes to more gracefully handle tasks that have been created via the API to invoke scripts. The UI does not know how to handle this new API capability.

## Experience

When a task that invokes a script is opened in the UI edit experience, the Flux edit panel is empty.

If the user adds some Flux to the text box, they _can_ save it successfully, and it changes the task to use that Flux instead of the script it used previously. This would not be a major problem, except there is no indication that this is happening and it gives the user the potential to damage the execution of tasks they have previously set up via the API to invoke scripts.

Note that activating/deactivating the task via the task run page works fine so no change is needed there. The issue is just when the editing experience is opened.

## Simple Suggestion

In the future, we will add proper support for editing tasks that reference scripts, but in the meantime, we should not allow users to open the edit experience at all for tasks that invoke scripts. We can simply disable the "EDIT TASK" button with a tooltip or error on click that says "Editing tasks that invoke scripts is not supported by the UI. Please use the API to edit this task".

### PR DESCRIPTION
**Pull request by glinton**
Closes #4978

This addresses an experience concern where a task is backed by a script by blocking users from saving flux if their task is backed by a script.

This fills the flux editing area with some notice text informing the user to use the api to interact with their script-backed task. It disables the save button as well. No impact on flux-backed task editing that I found.

image

### Checklist

Authors and Reviewer(s), please verify the following:

- [x] A PR description, regardless of the triviality of this change, that communicates the value of this PR
- [x] [Well-formatted conventional commit messages](https://www.conventionalcommits.org/en/v1.0.0/) that provide context into the change
- [ ] Documentation updated or issue created (provide link to issue/PR)
- [x] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed)
- [ ] Feature flagged, if applicable

### CODE REVIEW

File: src/types/tasks.ts [hunk 1]
```diff
@@ -3,6 +3,8 @@ import {NormalizedState, Run, RemoteDataState, LogEvent} from 'src/types'
 
 export interface Task extends Omit<ITask, 'labels'> {
   labels?: string[]
+  scriptID?: string
+  scriptParams?: string
```
[the remark this answers is not in the document]
Comment (glinton) [reply src/types/tasks.ts:7]: Not quite yet (unless you're ok with linking against a commit in an unmerged branch to openapi (in the past, someone inevitably points to HEAD), the openapi change was blocked by some decisions; one as i understood was this part of the UI.

**Comment by glinton** [comment]
> a lot of cruft in that component that would have been really nice to clean-up given the opportunity

Regarding the above comment, this "fix" doesn't eliminate the need for y'all to come and actually fix this. In my opinion, a hacky fix is worth more than a well designed buggy experience. So I'd say the opportunity is absolutely there with the added bonus of slightly more cruft to clean-up :wink:

File: src/types/tasks.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (glinton) [reply src/types/tasks.ts:7]: Just realized that the types are generated from the oss contract. Afaik, scripts are not present in oss, so linking to the openapi sha and regenerating still results in those two fields missing.

**Comment by appletreeisyellow** [comment]
@glinton merge the current master branch into your branch may help with passing `ci/circleci: monitor-ci-tests`

File: src/types/tasks.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (glinton) [reply src/types/tasks.ts:7]: Probably in the "correct" fix that will be done later by you/your team. In my external opinion, being able to handle the fields doesn't hurt when they aren't set, as even in cloud they are not always set. Here's the pr for reference https://github.com/influxdata/openapi/pull/349

**Pull request merged**